pandas/tests/io/test_sql.py

```diff
@@ -1153,14 +1161,14 @@ def setup_class(cls):
             msg = "{0} - can't connect to {1} server".format(cls, cls.flavor)
             pytest.skip(msg)
 
-    @pytest.fixture(autouse=True)
-    def setup_method(self, datapath):
-        self.setup_connect()
-
-        self._load_iris_data(datapath)
+    def load_test_data_and_sql(self):
         self._load_raw_sql()
         self._load_test1_data()
 
+    @pytest.fixture(autouse=True)
+    def setup_method(self, load_iris_data):
```

Comment:
Hmm I realize this is not something you have introduced but we seem to be ambiguously using both the classic xunit-style setup with a fixture and I think we should just use one or the other. Probably easiest to just remove the fixture decorator for now

Reply:
If we remove the fixture annotation this will cause the tests to fail in the current implementation. It seems like this is somewhat common to mix the old xunit style setup and fixtures. See @nicoddemus first comment on this issue. https://github.com/pytest-dev/pytest/issues/517

Reply:
But definitely agree we should maybe look at getting rid of the xunit style set up eventually :)

Reply:
What failure did that cause? In the provided link AFAICT it is showcasing an easy transition from old style to new, which is what I'm suggesting here as well

Reply:
The xunit setup is applied before any fixtures if we remove the annotation. For example in the test: `TestSQLApi.test_read_sql_iris`. The function `self.load_test_data_and_sql()` will be invoked before the `load_iris_data` fixture ( when `@pytest.fixture(autouse=True)` is not present ). 

Since `self.load_test_data_and_sql()` requires self.conn to be set, this will error. 

Hope that explains the issue clearly :)

Reply:
hey a friendly ping: @WillAyd are you happy to review ?  Hope my explanation addressed the concerns here

Reply:
Hmm OK understood on the issues you are facing but just seems like this can be simplified instead of adding new methods. Is it not possible to set the connection here instead of having it as a separate method?

Reply:
Thanks for getting back to me appreciate it - My thinking behind having it as a separate method was that the `_TestSQLAlchemy` class already has a separate method `setup_connect`. I did the same in `TestSQLiteFallback` and `_TestSQLApi`  so that my fixture `load_iris_data`  used across all these test classes could call the same method to do the required connection set up.  

Perhaps we need to consider a larger refactor of this file in order to make this set up just solely use fixtures, but with regards to getting rid of 100+ warnings in the 3.6 build mentioned #22338  this was the simplest and clearest change I could think of, let me know if you think otherwise, would be interested to see. Thanks as always for taking a look!

Reply:
Thanks for the explanation and your patience. I do see your point; could you open a follow up issue to refactor this module to use fixtures exclusively instead of the mix with inheritance?

Reply:
Sure i've raised here #22624. Mind adding the Tags, "Testing", "IO SQL" and perhaps "Medium Difficulty"